.github/workflows/build.yml

```diff
@@ -62,70 +96,157 @@ jobs:
       ## - uses: Swatinem/rust-cache@v2
 
       - name: Install cargo-llvm-cov
+        # Only need to run once
+        if: ${{ matrix.arch.arch == 'amd64' }}
```

Comment:
So we have this check everywhere... WDYT about having a separate workflow with all the new build steps that run first and handles ONLY the builds, then leave this existing one to just do everything else? 

This is kind of a nit, if it's too much of a PITA to deal with right now I understand

Reply:
That was my first implementation and it worked. But added ~10mins to the build. Things like lint and xtask commands build the code base,  so package install and builds are being duplicated.